src/main/java/com/opentable/db/postgres/embedded/PreparedDbProvider.java

```diff
@@ -76,12 +76,36 @@ private synchronized PrepPipeline createOrFindPreparer(DatabasePreparer preparer
 
     /**
      * Create a new database, and return it as a JDBC connection string.
-     * No two invocations will return the same database.
+     * NB: No two invocations will return the same database.
      */
     public String createDatabase() throws SQLException
     {
-        final DbInfo db = dbPreparer.getNextDb();
-        return getJdbcUri(db);
+        return getJdbcUri(createNewDatabase());
+    }
+
+    /**
+     * Create a new database, and return the backing info.
+     * This allows you to access the host and port.
+     * More common usage is to call createDatabase() and
+     * get the JDBC connection string.
+     * NB: No two invocations will return the same database.
+     */
+    public DbInfo createNewDatabase() throws SQLException
+    {
+       return dbPreparer.getNextDb();
+    }
+
+    /**
+     * Create a new Datasource given DBInfo.
+     * More common usage is to call createDatasource().
+     */
+    public DataSource createDataSourceFromDBInfo(final DbInfo dbInfo) throws SQLException
```

Comment:
Instead of exposing `DbInfo` to the end user (which is mostly an internal implementation detail), why not provide a separate factory method that accepts the relevant components of a `DbInfo` (namely `port`, `dbName`, and `user`) which will initialize the data source on the users behalf? Put differently, what's the motivation for exposing `DbInfo`?

Reply:
The *entire* point is to expose DBInfo. Because I need the port, user, password to create a DSN for Go for tests with vault.

Reply:
Reiterating what I said offline for posterity: I find the idea of allowing a test resource (like an embedded database) to escape the JVM to be mildly objectionable because it can very quickly lead to brittle, hard-to-debug code/tests. In the context of a junit rule, you have a precise lifecycle in which your tests can access the managed resource (eg in the body of a test/suite). If you allow those resources to escape, then you have to be very careful to cleanup the external resources in the correct order, lest you introduce random breakages and/or resource leaks because they haven't been cleaned up properly.

Again, the above objection is fairly mild. Realistically I don't see many people actually exporting database connection info from their tests, so my pushback is probably mostly hypothetical.